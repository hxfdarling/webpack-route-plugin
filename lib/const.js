exports.TYPE_BOOTSTRAP = "bootstrap";
exports.TYPE_ROUTES = "route";
