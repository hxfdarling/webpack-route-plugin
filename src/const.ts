export enum TYPE {
  BOOTSTRAP = 'bootstrap',
  ROUTES = 'route',
}
