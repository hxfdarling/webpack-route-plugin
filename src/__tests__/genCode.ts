import { join } from 'path';
import gen from '../genCode';
import { TYPE } from '../const';

const baseDir = join(__dirname, '../../demo');
const pagesDir = join(baseDir, 'pages');
process.env.NODE_ENV = 'development';
describe('gen code', () => {
  it('gen bootstrap', async () => {
    const code = gen({
      outputFile: join(baseDir, 'bootstrap.js'),
      baseDir: './test',
      pagesDir,
      indexFile: './index.js',
      type: TYPE.BOOTSTRAP,
    });
    expect(code).toMatchSnapshot();
  });
  it('gen route', async () => {
    const code = gen({
      pagesDir,
      outputFile: join(baseDir, 'routes.js'),
      type: TYPE.ROUTES,
    });
    expect(code).toMatchSnapshot();
  });
  it('gen route deep', async () => {
    const code = gen({
      pagesDir,
      outputFile: join(baseDir, 'routes.js'),
      type: TYPE.ROUTES,
      deep: true,
    });
    expect(code).toMatchSnapshot();
  });
});
