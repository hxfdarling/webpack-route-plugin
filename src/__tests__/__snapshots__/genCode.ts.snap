// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gen code gen bootstrap 1`] = `
"/* eslint-disable */
/**
 * 该文件为构建自动生成的启动文件，无需关注本文件，并且不要修改该文件内容
 * 文件之所以放在此处，是为了方便调试查看
 */
import bootstrap from './index.js';

const routes = [{
  name: 'xxx',
  sort: -2,
  component: () => import(\\"./../demo/pages/var.js\\")
}, {
  path: '/test',
  name: 'test',
  component: () => import(\\"./../demo/pages/empty.js\\")
}, {
  path: '/test2',
  component: () => import(\\"./../demo/pages/empty.js\\")
}, {
  path: '/',
  component: () => import(\\"./../demo/pages/home.js\\")
}, {
  name: 'test',
  path: '/work',
  component: () => import(\\"./../demo/pages/work/index.js\\")
}, {
  path: '/test',
  redirect: '/home',
  sort: 1000
}];

bootstrap(routes);
"
`;

exports[`gen code gen route 1`] = `
"/* eslint-disable */
/**
 * 该文件为构建自动生成的启动文件，无需关注本文件，并且不要修改该文件内容
 * 文件之所以放在此处，是为了方便调试查看
 */
const routes = [{
  name: 'xxx',
  sort: -2,
  component: () => import(\\"./demo/pages/var.js\\")
}, {
  path: '/test',
  name: 'test',
  component: () => import(\\"./demo/pages/empty.js\\")
}, {
  path: '/test2',
  component: () => import(\\"./demo/pages/empty.js\\")
}, {
  path: '/',
  component: () => import(\\"./demo/pages/home.js\\")
}, {
  name: 'test',
  path: '/work',
  component: () => import(\\"./demo/pages/work/index.js\\")
}, {
  path: '/test',
  redirect: '/home',
  sort: 1000
}];
export default routes;
    "
`;
